lib/rules/array-bracket-spacing.js

```diff
@@ -179,8 +179,10 @@ module.exports = {
 
             const first = sourceCode.getFirstToken(node),
                 second = sourceCode.getFirstToken(node, 1),
-                penultimate = sourceCode.getLastToken(node, 1),
-                last = sourceCode.getLastToken(node),
+                last = node.typeAnnotation
+                    ? sourceCode.getTokenBefore(node.typeAnnotation)
```

Comment:
I guess `sourceCode.getTokenAfter(/* last element */)` does not work if there is a trailing comma.